forem/forem
# [Search 2.0] Homepage query

## What type of PR is this? (check all applicable)


## Description

The homepage is currently powered by Elasticsearch used as a index (filtering, sorting and pagination). 
I added a combination of mini service classes to achieve the same with PostgreSQL:

- `Homepage::FetchArticles` is the entry point
- `Homepage::FetchTagFlares` solves the need of not having to call `article.flare_tag` for each article, which would result in one query per article
- `Homepage::ArticlesQuery` is where the filtering, sorting and pagination happens

## QA Instructions, Screenshots, Recordings

1. Click around and scroll on the homepage, both signed in and as an anonymous user
2. Activate the feature flag: `FeatureFlag.enable(:search_2_homepage)`
3. Do the same and check the results are alike

## Added tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Admin Guide](https://forem.gitbook.io/forem-admin-guide/), or
      [Storybook](https://storybook.forem.com/) (for Crayons components)
      or in a [forem.dev](http://forem.dev) post
      replace this line with details on why this change doesn't need to be
      shared_

## [optional] Are there any post deployment tasks we need to perform?

We should test this thoroughly on an internal Forem before releasing it on DEV, given how crucial it is :)
